onedal/basic_statistics/basic_statistics.py

```diff
@@ -62,52 +59,38 @@ def _get_onedal_params(self, dtype=np.float32):
             "result_option": options,
         }
 
-    def _compute_raw(self, data_table, weights_table, module, policy, dtype=np.float32):
-        params = self._get_onedal_params(dtype)
-
-        result = module.train(policy, params, data_table, weights_table)
 
-        options = self._get_result_options(self.options)
-        options = options.split("|")
+class BasicStatistics(BaseBasicStatistics):
+    """
+    Basic Statistics oneDAL implementation.
+    """
 
-        return {opt: getattr(result, opt) for opt in options}
+    def __init__(self, result_options="all", algorithm="by_default"):
+        super().__init__(result_options, algorithm)
 
-    def _compute(self, data, weights, module, queue):
-        policy = self._get_policy(queue, data, weights)
+    def fit(self, data, sample_weight=None, queue=None):
+        policy = self._get_policy(queue, data, sample_weight)
 
         if not (data is None):
             data = np.asarray(data)
```

Comment:
asarray() is not compatible with csr data. Even though this PR does not deal with sparse data, it will be better to use check_array function.